InterruptServer.rb

```diff
@@ -0,0 +1,212 @@
+require 'socket'
+require 'json'
+
+class InterruptServer
+
+	CHAT_LENGTH = 45 # length of chat text user can see at one time
+	TICK_LENGTH = 2 # how many seconds to wait before 'moving chat text left'
+	MAX_MSG_LENGTH = 1024 # max length of incoming message read
+	
+	def initialize(host, port, num_colors=5)
+		@server = UDPSocket.new
+		@server.bind(host, port)
+
+		@clients = {}
+		@colors_available = [*0..num_colors-1]
```

Comment:
That's a good point; perhaps I should rename it. It could be called "voice", or "position" for instance. From the point of view of the server, it is a finite list of items that can be assigned to at most one client at a time.